modules/dnn/src/tensorflow/tf_graph_simplifier.cpp

```diff
@@ -832,7 +837,7 @@ void RemoveIdentityOps(tensorflow::GraphDef& net)
     }
 }
 
-Mat getTensorContent(const tensorflow::TensorProto &tensor, bool copy)
+Mat getTensorContentRefUnaligned(const tensorflow::TensorProto& tensor)
```

Comment:
Mat's `data` pointer must be aligned on base type requirements.
Otherwise it is impossible to use it in OpenCV algorithms on some platforms.

> Mat getTensorContentRefUnaligned

So this function is wrong by design. Mat should not contain "unaligned" pointer.

---

I believe, it is better to fix existed implementation.
Check pointer alignment and reallocate Mat for such case.
Parameter `copy` should be renamed to `forceCopy` due to behavior changes.

Reply:
The code's structure does not allow realigning the offending `Mat`, because PReLUSubgraph's finalize method depends on having write access to the misaligned data block.

My motivation with the PR was focused on resolving the crash without changing semantics.

I do not sufficiently understand the remainder of the implementation at this point to reengineer its complete structure and keep the functionality of PReLUSubgraph by propagating an aligned copy of the tensor content. Instead, I have taken special care that the previously broken code now behaves correctly and as probably intended by the author by accessing the offending `Mat` via memcpy only.

To address your last sentence regarding renaming `copy` => `forceCopy`: You cannot safely keep the existing behavior, it goes against what you said in the beginning where a `Mat` must be aligned. No copy equals not aligned, since raw Protobuf data goes into that function.

<hr/>

Please let me know how to proceed here. I could have it return a `void*` instead, which would technically fulfill your requirement of not having an unaligned Mat as return value?